emu/src/cpu/arm7tdmi.rs

```diff
@@ -1094,6 +1098,33 @@ impl Arm7tdmi {
         }
     }
 
+    pub(crate) fn multiple_load_store(
+        &mut self,
+        load_store: LoadStoreKind,
+        base_register: u16,
+        register_list: u16,
+    ) -> Option<u32> {
+        match load_store {
+            LoadStoreKind::Store => unimplemented!("multiple store"),
+            LoadStoreKind::Load => {
+                let base_address = self.registers.register_at(base_register as usize);
+                for r in 0..=7 {
+                    if register_list.get_bit(r) {
+                        let address = base_address + (r as u32 * 4);
```

Comment:
I'm not sure this is correct, imagine a register list like this: `0b00000010` and `R0 = 0x100`. We should then read at `0x100` and write in `R1`. But with this logic we read as `0x104`, right?